frontend/nginx/nginx.conf

```diff
@@ -23,10 +23,26 @@ http {
     gzip  on;
     include /etc/nginx/conf.d/*.conf;
 
+    upstream shards {
+        server shard0;
+        server shard1;
+    }
+
     server {
         root /app;
         index index.html;
         listen 8000;
         listen [::]:8000;
+
+        location /submit {
+            proxy_pass http://shards/submit;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header Host $host;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+
+            proxy_http_version 1.1;
+            proxy_set_header Upgrade $http_upgrade;
+            proxy_set_header Connection "upgrade";
+        }
```

Comment:
Why mixing it with frontend instead of adding these on host's Nginx instead just like it is done already?

Reply:
Because shards won't be accessible via docker dns/hosts in host nginx

Reply:
It will be if instead of exposing one shard on `8081` and another on `8082` and create two upstreams in host Nginx's config.

I really dislike that we break abstractions in an ugly way here.

Reply:
Removed configuration on the frontend. I'll add shard balancing on the host nginx